soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4620,6 +4641,69 @@ void CreateScrubMessages() {
     }
 }
 
+void CreateRupeeMessages() {
+    CustomMessageManager* customMessageManager = CustomMessageManager::Instance;
+    customMessageManager->AddCustomMessageTable(Randomizer::rupeeMessageTableID);
+    const std::vector<u8> rupees = { TEXT_BLUE_RUPEE, TEXT_RED_RUPEE, TEXT_PURPLE_RUPEE, TEXT_HUGE_RUPEE };
+    std::string rupeeText = " ";
+    for (u8 rupee : rupees) {
+        switch (rupee) {
+            case TEXT_BLUE_RUPEE:
+                rupeeText = "\x05\x03 5 #RUPEE#\x05\x00";
+                break;
+            case TEXT_RED_RUPEE:
+                rupeeText = "\x05\x01 20 #RUPEE#\x05\x00";
+                break;
+            case TEXT_PURPLE_RUPEE:
+                rupeeText = "\x05\x05 50 #RUPEE#\x05\x00";
+                break;
+            case TEXT_HUGE_RUPEE:
+                rupeeText = "\x05\x06 200 #RUPEE#\x05\x00";
+                break;
+        }
+        customMessageManager->CreateMessage(Randomizer::rupeeMessageTableID, rupee,
+            { TEXTBOX_TYPE_BLACK, TEXTBOX_POS_BOTTOM,
+                "You found" + rupeeText + " !",
+                "Du hast" + rupeeText + "  gefunden!",
+                "Vous obtenez" + rupeeText + " !"
+        });
+    }
+}
+
+std::string Randomizer::InsertRupeeName(std::string message, int language) {
+    int randomIndex;
+    std::string replaceWith;
+    switch (language) {
+        case LANGUAGE_ENG:
+            randomIndex = rand() % (sizeof(englishRupeeNames) / sizeof(englishRupeeNames[0]));
+            replaceWith = englishRupeeNames[randomIndex];
+            break;
+        case LANGUAGE_GER:
+            randomIndex = rand() % (sizeof(germanRupeeNames) / sizeof(germanRupeeNames[0]));
+            replaceWith = germanRupeeNames[randomIndex];
+            break;
+        case LANGUAGE_FRA:
+            randomIndex = rand() % (sizeof(frenchRupeeNames) / sizeof(frenchRupeeNames[0]));
+            replaceWith = frenchRupeeNames[randomIndex];
+            break;
+    }
+    std::string replaceString = "#RUPEE#";
+    size_t pos = message.find(replaceString);
+    size_t len = replaceString.length();
+    message.replace(pos, len, replaceWith);
+    CustomMessageManager::Instance->FormatCustomMessage(message);
+    return message;
+}
+
+CustomMessageEntry Randomizer::GetRupeeMessage(u16 rupeeTextId) {
```

Comment:
if the other one is renamed, this might make sense as `GetRandomizedRupeeMessageEntry`